js/fibonacci.js

```diff
@@ -0,0 +1,178 @@
+            return arrayToSort.sort(function (firstObj, lastObj) { return lastObj.createdDate - firstObj.createdDate });
+    }
+};
+
+// =================================================================
+async function getFibonacciFromServer(num) {
+    try {
+        const FIBONACCI_URL = `http://localhost:5050/fibonacci/${num}`;
+        const response = await fetch(FIBONACCI_URL);
+        if (response.ok) {
+            const data = await response.json();
+            return data.result;
+        } else {
+            throw response;
+        }
+    }
+    catch (error) {
+        const err = await error.text();
+        return err;
+    }
+};
+
+// function getFibonacciFromServer(num) {
+//     let FIBONACCI_URL = `http://localhost:5050/fibonacci/${num}`;
+//     return fetch(FIBONACCI_URL)
+//         .then(function (response) {
+//             if (response.ok) {
+//                 return response.json();
+//             } else {
+//                 return response.text();
+//             }
+//         })
+//         .then(function (data) {
+//             if (typeof data === 'object') {
+//                 return data.result;
+//             }
+//             return data;
+//         })
+//         .catch(function (error) {
+//             return error.text();
+//         })
+// };
+
+function getFibonacci(x) {
+    if (x <= 0) {
+        return 0;
+    } else if (x === 1) {
+        return 1;
+    } else {
+        let arrOfFibonacci = [0, 1];
+        for (let index = 2; index < x + 2; index++) {
+            arrOfFibonacci[index] = arrOfFibonacci[index - 2] + arrOfFibonacci[index - 1];
+        }
+        return arrOfFibonacci[x];
+    }
+};
+
+// =================================================================
+
+async function getFibonacciResults() {
+    const FIBONACCI_RESULT_URL = `http://localhost:5050/getFibonacciResults`;
+    const response = await fetch(FIBONACCI_RESULT_URL);
+    const data = await response.json();
+    return data.results;
+};
+
+// function getFibonacciResults() {
+//     const FIBONACCI_RESULT_URL = `http://localhost:5050/getFibonacciResults`;
+//     return fetch(FIBONACCI_RESULT_URL)
+//         .then(function (response) {
+//             return response.json();
+//         })
+//         .then(function (data) {
+//             let arrayOfResponses = data.results;
+//             arrayOfResponses.sort(function (firstObj, lastObj) { return lastObj.createdDate - firstObj.createdDate });
+//             return arrayOfResponses;
+//         })
+// };
+
+function displayResults(array) {
+    let itemDate;
+    let currentItem;
+    const desiredLength = 10;
+    for (let i = 0; i < desiredLength; i++) {
+        itemDate = new Date(array[i].createdDate);
+        currentItem = array[i];
+        calcRes.innerHTML += `<span class="fs-5 border-bottom pb-3 border-secondary">&nbsp;The Fibonacci of <b>${currentItem.number}</b> is <b>${currentItem.result}</b>. Calculated at: ${itemDate}</span>`;
```

Comment:
Manipulating the DOM inside a loop is not a good practice. Instead, create a container and append the items insie the loop to the continaer, and append the container to the DOM after the loop is finished. Anoter option is to use documentFragment:
https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment